jsonschema-core.md

```diff
@@ -880,29 +882,50 @@ vocabulary's keywords.
 Meta-schema authoring is an advanced usage of JSON Schema, so the design of
 meta-schema features emphasizes flexibility over simplicity.
 
+#### Dialect Determination
+
+When evaluation encounters a new schema resource (i.e. the lexical scope
+changes), the first task is to determine the dialect used by the schema. To do
+this, implementations MUST determine the dialect using the following prioritized
+steps.
+
+1. The `$schema` keyword - Implementations MUST process the schema according to
+   the dialect it declares.
+2. `application/schema+json` media type with a `schema` parameter -
+   Implementations which support media type parameter inputs MUST process the
+   schema according to the dialect the parameter declares. A media type will
+   generally only be available if the schema has been retrieved from an external
+   source and only applies to the document root.
+3. Parent dialect - An embedded schema resource which does not itself contain a
+   `$schema` keyword MUST be processed using the same dialect as the schema
+   which contains it.
```

Comment:
I think this concept also applies to schemas embedded in other types of documents like OpenAPI. I'm not sure how this might be reworded to include that case, or if it should be considered a different case, or if that case should be addressed elsewhere entirely.

Reply:
We can say something like:

If the schema is embedded in a larger document (such as OpenAPI), the semantics for determining the dialect may be specified by that document's specification. Unless explicitly dictated by that document's specification, the presence of a `$schema` keyword shall always override the dialect imposed by the containing document.

Reply:
I like that @karenetheridge.  I'll try to work that in.

Reply:
I agree. Currently, the OpenAPI spec allows you to define a default dialect.

@karenetheridge What's the use case you see for allowing a containing document specification to override `$schema` for it's contained schemas?

Reply:
No, the other way around - the openapi spec has language about following `jsonSchemaDialect` to set the dialect for all contained schemas, but if one of those schemas has a `$schema` keyword in it, that overrides the jsonSchemaDialect. (Unless OpenAPI wanted to add a rule saying that `$schema` is no good here.)

Reply:
Right, so that falls in line with the precedence order here.  `$schema` first, then media type (if it exists), then parent document, then config.

I think Ben was wondering about the edge case where a document wanted to override `$schema`.  I'm not sure we can define requirements for that.